lib/pyfrc/test_support/pytest_plugin.py

```diff
@@ -59,6 +59,9 @@ def pytest_runtest_setup(self):
                        "If your robot class has an __init__ function, it must call super().__init__()!"
     
     def pytest_runtest_teardown(self, nextitem):
+        # Let any child threads run in realtime to allow cancelling if it
+        # has been implemented.
+        self._fake_time.freerun_children()
```

Comment:
I can reliably detect still-running threads and throw an exception.
A side effect is that the threads throw other exceptions anyway (like trying to access the HAL after it has been torn down), and subsequent tests also fail because the test harness is no longer initialised properly.
The good bit is that you do get an exception spelling out the root cause - unterminated threads.
Is this behaviour acceptable?